File: lua/plugins/indent-blankline.ts
import { LazyPlugin } from "../../ambient/lazy";
import { getGlobalConfiguration } from "../helpers/configuration";
import { useExternalModule } from "../helpers/module/useModule";

const plugin: LazyPlugin = {
  1: 'lukas-reineke/indent-blankline.nvim',
  version: "^3",
  config: function(this: void) {
    const ibl = useExternalModule<{
      setup: (this: void, opts?: unknown) => void
    }>("ibl");
    ibl.setup({
      scope: {
        enabled: getGlobalConfiguration().packages.hlchunk?.enabled ? false : true
      }
    });
  }
};
export { plugin as default };

File: lua/plugins/indent-blankline.lua
return {
  {
    "lukas-reineke/indent-blankline.nvim",
    main = "ibl", -- The entry point for the plugin
    event = { "BufReadPost", "BufNewFile" },
    opts = {
      indent = {
        char = "▎", -- You can use "│", "┆", "┊", etc.
        tab_char = "▎",
      },
      scope = {
        enabled = true, -- Highlights the current code block
        show_start = false,
        show_end = false,
      },
      exclude = {
        filetypes = {
          "help",
          "alpha",
          "dashboard",
          "neo-tree",
          "Trouble",
          "lazy",
          "mason",
          "notify",
          "toggleterm",
          "lazyterm",
        },
      },
    },
  },
}
